### PR TITLE
✨ YakShaver Blogging

### PR DESCRIPTION
**Pull request by joshbermanssw**
This PR addresses 2 different PBIs in the YakShaver backlog: 
-[✨ Create Blog Index Page with Themed Cards](https://github.com/SSWConsulting/SSW.YakShaver/issues/788)
- [✨ Create Blog Inner Page with Enhanced Features](https://github.com/SSWConsulting/SSW.YakShaver/issues/783)

FIGMA CONTEXT: https://www.figma.com/proto/Y2klH0n3k7H1NUBPsG8epQ/SSW-Product-Sites?node-id=358-326&node-type=canvas&t=TT4dSj8Eg2A1m8m3-0&scaling=scale-down-width&content-scaling=fixed&page-id=226:536&hide-ui=1

I have: 
- Created a new collection for the blogs 
- Created a new route from the root of each {{ product }} called /blog
- The /blog page will use a custom gql query to query all blogs and then filter by those starting with /{{ product}} 
- The /blog page will populate the page with cards displaying previews of each blog post as per figma 
- The /blog page will only display a preview of the content of the markdown body -> i have set this as the first 3 sentences but upon reflection this maybe should a character limit? 
- Created a dynamic route for each specific blog post called blog/[slug] which takes in the filename and uses the auto generated gql query to query for that specific {{ product }} / {{ filename }} and display the full blog page

Screenshot 2024-10-11 at 4 13 03 pm
**Figure: Posts Page with some Dummy Posts**

### CODE REVIEW

File: app/[product]/blog/[slug]/page.tsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+export default async function BlogPost({ params }: BlogPostProps) {
+  const { slug, product } = params;
+
+  const data = await getBlogPost(product, slug);
+  if (!data) {
+    return notFound();
+  }
+
+  console.log("data here", data);
```
Comment (bradystroud) [inline app/[product]/blog/[slug]/page.tsx:23]: 🚮

File: utils/extractBlurbAsTinaMarkdownContent.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+          }
+        }
+        if (paragraphNode.children.length > 0) {
+          blurb.children.push(paragraphNode);
+        }
+        if (sentenceCount >= sentenceLimit) {
+          break;
+        }
+      }
+    }
```
Comment (bradystroud) [inline utils/extractBlurbAsTinaMarkdownContent.ts:50]: Think about if we could do the same thing on Tina.io

**Review by bradystroud — CHANGES_REQUESTED** [review]
As per our conversation, there are a few things that could be improved (e.g. adding hover effects) but Betty will pick these up later

File: tina/collectionSchema/blog.tsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    {
+      type: 'string',
+      name: 'sswPeopleLink',
+      label: 'Author SSW People Link'
+    },
+    {
+      type: 'string',
+      name: 'readLength',
+      label: 'Read (time) length'
+    },
```
[the remark this answers is not in the document]
Comment (joshbermanssw) [reply tina/collectionSchema/blog.tsx:46]: done

File: app/[product]/blog/page.tsx [hunk 4]
```diff
@@ -0,0 +1,42 @@
+import { notFound } from 'next/navigation';
+import { getBlogsForProduct } from '../../../utils/fetchBlogs'; // Updated import
+import InteractiveBackground from '../../../components/shared/Background/InteractiveBackground';
+import NavBarServer from '../../../components/shared/NavBarServer';
+import FooterServer from '../../../components/shared/FooterServer';
+import BlogPageClient from '../../../components/shared/BlogPageClient';
+
+interface BlogPageProps {
```
[the remark this answers is not in the document]
Comment (joshbermanssw) [reply app/[product]/blog/page.tsx:8]: done

File: app/[product]/blog/page.tsx [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+      return notFound();
+    }
+
+    return (
+      <div className="flex flex-col min-h-screen">
+        <InteractiveBackground />
+        <NavBarServer product={product} />
+
+        <div className="flex-grow">
+          <BlogPageClient
```
[the remark this answers is not in the document]
Comment (joshbermanssw) [reply app/[product]/blog/page.tsx]: done

File: utils/extractBlurbAsTinaMarkdownContent.ts (hunk 2 above)
Comment (joshbermanssw) [reply utils/extractBlurbAsTinaMarkdownContent.ts:50]: created a spike https://github.com/tinacms/tina.io/issues/2329

**Review by bradystroud — APPROVED** [review]

File: tina/collectionSchema/blog.tsx [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+    {
+      type: "rich-text",
+      name: "body",
+      label: "Body",
+      isBody: true,
+      parser: {
+        type: "mdx",
+      },
```
[the remark this answers is not in the document]
Comment (bradystroud) [reply tina/collectionSchema/blog.tsx:55]: Please make a PBI if you dont plan on fixing this today

[the remark this answers is not in the document]
Comment (joshbermanssw) [reply tina/collectionSchema/blog.tsx:55]: Following up with @Ben0189 then will take action 👍

**Pull request merged**